### PR TITLE
Task migration unit tests

### PR DESCRIPTION
**Pull request by tbak**
### Description of the Change

This commit includes also a major refactoring of the data generation framework
for job, agent and eviction components.

### CODE REVIEW

File: titus-api/src/main/java/com/netflix/titus/api/agent/model/AgentFunctions.java [hunk 1]
```diff
@@ -47,4 +48,18 @@ private AgentFunctions() {
...
+
+    public static Function<AgentInstanceGroup, AgentInstanceGroup> withId(String newId) {
+        return ig -> ig.toBuilder().withId(newId).build();
+    }
+
+    public static Function<AgentInstanceGroup, AgentInstanceGroup> inState(InstanceGroupLifecycleState state) {
+        return ig -> ig.toBuilder().withLifecycleStatus(
+                InstanceGroupLifecycleStatus.newBuilder()
+                        .withState(state)
+                        .withTimestamp(System.currentTimeMillis())
```
Comment (corindwyer) [inline titus-api/src/main/java/com/netflix/titus/api/agent/model/AgentFunctions.java]: Shouldn't this be based on the clock abstraction?

File: titus-api/src/main/java/com/netflix/titus/api/agent/model/AgentFunctions.java [hunk 2]
```diff
@@ -47,4 +48,18 @@ private AgentFunctions() {
...
+                        .withDetail("No reason given")
```
Comment (corindwyer) [inline titus-api/src/main/java/com/netflix/titus/api/agent/model/AgentFunctions.java]: Shouldn't this force a reason in the contract as there is generally always a reason to doing the transition.

File: titus-api/src/main/java/com/netflix/titus/api/agent/model/AgentFunctions.java (hunk 1 above)
Comment (tbak) [reply titus-api/src/main/java/com/netflix/titus/api/agent/model/AgentFunctions.java]: Fixed

File: titus-api/src/main/java/com/netflix/titus/api/agent/model/AgentFunctions.java (hunk 2 above)
Comment (tbak) [reply titus-api/src/main/java/com/netflix/titus/api/agent/model/AgentFunctions.java]: I created `AgentTestFunctions` and moved this method there.

**Pull request merged**